### ISSUE TITLE
Main Window doesn't remember open folders

### ISSUE THREAD

**Issue by cbrchrt** [autokey/autokey#144]
Classification:
UI/Usability

Reproducibility:
Always

Summary
When opening the main window I have to

Steps to Reproduce
Select 'Show Main Window' on the Autokey icon in the desktop panel bar.

Expected Results
Main window should open with the folders open from the previous time.

Actual Results
All folders are closed.

Version
0.94.1

Installed via: PPA

Distro: Ubuntu 18.04

Notes
Re-opening as suggested.

**Comment by luziferius** [comment]
Side note: This happens both in the Qt and GTK version.

I think this happens because the GUI window is destroyed and completely removed from memory when closed and recreated from scratch when opened again. This loses the current state, including opened tree elements (folders). (The Qt code looks like working like this…)

@cbruchert Would it be sufficient for you to remember the opened folders during the runtime (from AutoKey start, across window closes/re-opens, until exit) or across program restarts (this includes OS/PC restarts)?

**Comment by sporkwitch** [comment]
@luziferius worth noting, he would be working from the GTK version; I haven't included the Qt in the 18.04 PPA because it lacks the Qt deps

For my two cents, I think during a session would be sufficient.

**Comment by cbrchrt** [comment]
@luziferius Maybe having it remember across program restarts would be nice too, but for me the most important is during the session.

**Comment by luziferius** [comment]
Session-only allows the easiest solution, because there is no persistent storage involved that can desync from the file system content.

**Comment by josephj11** [comment]
@luziferius 's repo is still there with all these PRs in it. IDK if any of this made it into develop.

There's another related issue to this. #703

**Comment by Elliria** [comment]
I agree that it should remember its state during an open session, but having it remember its state from session to session should be optional. I would opt out of that, because I'd find it disorienting for the program to open in anything but its default state on its first load of a session.

**Comment by josephj11** [comment]
@Elliria Persistence for the duration of the session makes the most sense to me. All these commits are so old though that they could only be used as suggestions at this point.

I think this should be fixed, but that it's pretty low priority ATM.

**Comment by Elliria** [comment]
Yep. I agree. They're popping up in my [CodeTriage](https://www.codetriage.com/) emails, so I check them to see if they're easily fixable or comment on them or label them, as the case may be.
